code/modules/mob/living/silicon/silicon.dm

```diff
@@ -353,3 +354,26 @@
 	..()
 	if(cameraFollow)
 		cameraFollow = null
+
+/mob/living/silicon/Move(newloc, direct)
+	..(newloc,direct)
+	if (underdoor)
+		if (!(layer == UNDERDOOR))
+			//We must have used hide, or had our layer changed by something else. cancel this processing
+			underdoor = 0
+
+		underdoor = 0
+		for (var/obj/machinery/door/D in loc)
+			if (D.hashatch)
+				underdoor = 1
+				break
+
+		if (!underdoor)
+			spawn(3)//A slight delay to let us finish walking out from under the door
+				layer = initial(layer)

```

Comment:
This doesn't account for a drone who's hiding, and will reset it.

Making the drone player have to re-toggle hide() every time he moves under a door is going to be annoying, so please make it account for it.